Microsoft.HealthVault.Fhir/Transformers/MedicationRequestToHealthVault.cs

```diff
@@ -0,0 +1,193 @@
+
+
+
+using System;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Codes.HealthVault;
+using Microsoft.HealthVault.Fhir.Codings;
+using Microsoft.HealthVault.Fhir.FhirExtensions.Helpers;
+using Microsoft.HealthVault.ItemTypes;
+using Microsoft.HealthVault.Thing;
+using UnitsNet.Extensions.NumberToDuration;
+using FhirMedication = Hl7.Fhir.Model.Medication;
+using HVMedication = Microsoft.HealthVault.ItemTypes.Medication;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class MedicationRequestToHealthVault
+    {
+        public static ThingBase ToHealthVault(this MedicationRequest medicationRequest)
+        {
+            var fhirMedication = ExtractEmbeddedMedication(medicationRequest);
+
+            var hvMedication = fhirMedication.ToHealthVault() as HVMedication;
+
+            ResourceReference agentReference = medicationRequest.Requester?.Agent;
+            if (agentReference != null && agentReference.IsContainedReference)
+            {
+                var practitioner = ExtractEmbeddedPractitioner(medicationRequest);
+                var prescription = new Prescription(practitioner.ToHealthVault());
+                if (medicationRequest.AuthoredOnElement != null)
+                {
+                    var dt = medicationRequest.AuthoredOnElement.ToDateTimeOffset();
+                    prescription.DatePrescribed = new ApproximateDateTime(
+                            new ApproximateDate(dt.Year, dt.Month, dt.Day),
+                            new ApproximateTime(dt.Hour, dt.Minute, dt.Second, dt.Millisecond));
+                }
+
+                MedicationRequest.DispenseRequestComponent dispenseRequest = medicationRequest.DispenseRequest;
+                if (dispenseRequest != null)
+                {
+                    var numerator = dispenseRequest.Quantity;
+
+                    if (numerator != null)
+                    {
+                        var structuredMeasurement = new StructuredMeasurement
+                        {
+                            Value = (double)numerator.Value,
+                            Units = CodeToHealthVaultHelper.CreateCodableValueFromQuantityValues(numerator.System,
+                                   numerator.Code, numerator.Unit)
+                        };
+                        prescription.AmountPrescribed = new GeneralMeasurement();
+                        prescription.AmountPrescribed.Structured.Add(structuredMeasurement);
+                    }
+
+                    if (dispenseRequest.NumberOfRepeatsAllowed.HasValue)
+                    {
+                        prescription.Refills = dispenseRequest.NumberOfRepeatsAllowed;
+                    }
+                    else
+                    {
+                        prescription.Refills = 0;
```

Comment:
>     prescription.Refills = 0; [](start = 19, length = 30)

Refills is a Nullable Int, you should not set to 0 if there was no value for NumberOfRepeatsAllowed.

Reply:
Well, [NumberOfRepeatsAllowed](https://www.hl7.org/fhir/medicationrequest-definitions.html#MedicationRequest.dispenseRequest.numberOfRepeatsAllowed) allows for only [positiveInt](https://www.hl7.org/fhir/datatypes.html#positiveInt). Absence of the element implied  0 ( see [Fhir chat](https://chat.fhir.org/#narrow/stream/implementers/subject/MedicationRequest.20Disallow.20refills/near/96308)).

The newer build @ [build.fhir.org](http://build.fhir.org/medicationrequest-definitions.html#MedicationRequest.dispenseRequest.numberOfRepeatsAllowed) seems to allow 0. Should we change to this implementation?